server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistAnalysisPropertiesStep.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.task.projectanalysis.step;
+
+import org.sonar.core.util.UuidFactoryFast;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.component.AnalysisPropertyDto;
+import org.sonar.server.computation.task.projectanalysis.analysis.AnalysisMetadataHolder;
+import org.sonar.server.computation.task.projectanalysis.batch.BatchReportReader;
+import org.sonar.server.computation.task.step.ComputationStep;
+
+/**
+ * Persist analysis properties
+ * Only properties starting with "sonar.analysis" will be persisted in database
+ */
+public class PersistAnalysisPropertiesStep implements ComputationStep {
+
+  private final DbClient dbClient;
+  private final AnalysisMetadataHolder analysisMetadataHolder;
+  private final BatchReportReader reportReader;
+  private static final String SONAR_ANALYSIS = "sonar.analysis.";
+
+  public PersistAnalysisPropertiesStep(DbClient dbClient, AnalysisMetadataHolder analysisMetadataHolder,
+    BatchReportReader reportReader) {
+    this.dbClient = dbClient;
+    this.analysisMetadataHolder = analysisMetadataHolder;
+    this.reportReader = reportReader;
+  }
+
+  @Override
+  public void execute() {
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      reportReader.readContextProperties().forEachRemaining(
```

Comment:
I'd suggest to read first context properties (with filtering on `SONAR_ANALYSIS` prefix) to only open a DB connection when there is at least one property to persist.

This will load a bit of data into memory but in most case, that will save opening a DB connection.
Remember, specifying such properties is quite an advanced feature. We can therefor safely expect there will rarely be any such property to persist.